lib/spack/spack/build_environment.py

```diff
@@ -583,10 +584,13 @@ def set_package_py_globals(pkg, context: Context = Context.BUILD):
     # Put spack compiler paths in module scope. (Some packages use it
     # in setup_run_environment etc, so don't put it context == build)
     link_dir = spack.paths.build_env_path
-    module.spack_cc = os.path.join(link_dir, pkg.compiler.link_paths["cc"])
-    module.spack_cxx = os.path.join(link_dir, pkg.compiler.link_paths["cxx"])
-    module.spack_f77 = os.path.join(link_dir, pkg.compiler.link_paths["f77"])
-    module.spack_fc = os.path.join(link_dir, pkg.compiler.link_paths["fc"])
+    try:
```

Comment:
This looks a bit too generic. Can't you catch a specific error? Maybe refactor to `compiler = get_compiler(pkg)` and then `if compiler: module.spack_cc = ...` etc?

Reply:
Per DM, we need to check how to deal with `mpi` implementations, and similar specs that have a real run dependency on the compiler.